packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright 2018-2021 BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers'
+import type {
+  Deposit,
+  IDidIdentifier,
+  IIdentity,
+  KeyringPair,
+  KeystoreSigner,
+  SubmittableExtrinsic,
+  VerificationKeyType,
+} from '@kiltprotocol/types'
+
+import type { SignerOptions } from '@polkadot/api/types'
+import type { Null, Option, Struct } from '@polkadot/types'
+import type { AccountId, Extrinsic } from '@polkadot/types/interfaces'
+import type { AnyNumber } from '@polkadot/types/types'
+import type { HexString } from '@polkadot/util/types'
+
+import type { DidAuthorizationOptions, FullDidDetails } from '../DidDetails'
+import { getKiltDidFromIdentifier, parseDidUri } from '../Did.utils.js'
+
+// TODO: update with string pattern types once available
+type AccountAddress = IIdentity['address']
```

Comment:
I would call this `KiltAccountAddress` or similar, to distinguish it from the generic `AccountId`, which can be from any other chain.

Reply:
I don't think we care how the address is encoded, that should all go through actually

Reply:
Going back to this issue, it now appears that if you give a generic Substrate address which is automatically encoded by the SDK, and then try to query the DID for that address, it fails as the address is not automatically encoded upon querying. There is an integration test case that is currently failing because of this.